arangod/Scheduler/SchedulerFeature.cpp

```diff
@@ -420,15 +424,37 @@ extern "C" void c_exit_handler(int signal, siginfo_t* info, void*) {
 }
 
 extern "C" void c_hangup_handler(int signal, siginfo_t* info, void*) {
-  if (signal == SIGHUP) {
-    std::string sender =
-        info ? std::to_string(info->si_pid) : std::string("unknown");
-    LOG_TOPIC("33eae", INFO, arangodb::Logger::FIXME)
-        << "hangup received, about to reopen logfile (sender pid " << sender
-        << ")";
-    LogAppender::reopen();
-    LOG_TOPIC("23db2", INFO, arangodb::Logger::FIXME)
-        << "hangup received, reopened logfile";
+  TRI_ASSERT(signal == SIGHUP);
+
+  // id of process that issued the SIGHUP
+  pid_t processIdRequesting = info ? info->si_pid : 0;
+  // the expected process id that we want to see
+  pid_t processIdExpected = ::processIdUnspecified;
+
+  // only set log rotate request if we don't have one queued already. this
+  // prevents duplicate execution of log rotate requests.
+  // if the CAS fails, it doesn't matter, because it means that a log rotate
+  // request was already queued
+  ::processIdRequestingLogRotate.compare_exchange_strong(processIdExpected,
+                                                         processIdRequesting);
```

Comment:
Why do you attach this to the sender process id? Would it be sufficient to use a simple `isLogRotateScheduled` flag?